blogs/2022/03/28/increase-productivity-with-containers.md

```diff
@@ -0,0 +1,89 @@
+
+Containers first emerged in the virtual world for this exact same benefit: a streamlined way to ship products. They provide a way to ship your software as a unit, including the binaries, dependencies, and configuration files.  This reduces the time spent on configuration overhead, as the container packages everything that the software needs to be run. The container can then be easily deployed from environment to environment. It’s [predicted](https://www.gartner.com/en/newsroom/press-releases/2020-06-25-gartner-forecasts-strong-revenue-growth-for-global-co) that by 2023, more than 75% of global organizations will be running containerized applications in production, up from less than 30% in 2020.
+
+Traditionally, containers come into play when the code is ready to be deployed to production. While this model certainly streamlines the end of the software development lifecycle, it doesn’t do much for developers when they’re actually writing and testing the code. To solve this gap, container-based development emerged.
+
+## Containers for developers
+
+The idea behind container-based development is to introduce containers at the very start of the development process.  Developers can then do all coding and testing in environments that are consistent with both the production environment and other dev environments. Because of this consistency across environments, time spent troubleshooting different code behavior from one environment to another can be virtually eliminated.
+
+This brings us to the concept of a development container: a container that runs a full-featured development environment. A dev container houses its own app and dependencies, such as the required tools, libraries, and runtimes. In the diagram below, you can see that these dependencies exist in the containers and not on the host machine, which means that you can seamlessly switch between tech stacks in no time at all.
+
+![A diagram showing containers with different tech stacks](container-diagram.png)
+Image Source: Development Containers in Education with Visual Studio Code
+
+To provide a way to create and connect to dev containers, VS Code unveiled the [Remote -  Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) in 2019. This extension provides the illusion of local development by leveraging the full power of dev containers, all while never leaving the comfort of VS Code. With over 11 million installs of the extension, it got us thinking: what if you could have a dev container that is hosted in the cloud?
```

Comment:
"provides the illusion of local development" might sound like users are in the cloud or otherwise remote machine (common misconception). Maybe something like this would help:

```suggestion
To provide a way to create and connect to dev containers, VS Code unveiled the [Remote -  Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) in 2019. This extension enhances local development by leveraging the full power of dev containers, all while never leaving the comfort of VS Code. With over 11 million installs of the extension, it got us thinking: what if you could have a dev container that is hosted in the cloud?
```